lib/resourcebuilder/podspec.go

```diff
@@ -0,0 +1,47 @@
+package resourcebuilder
+
+import (
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+// updatePodSpecWithProxy mutates the input podspec with proxy env vars for all init containers and containers
+// matching the container names.
+func updatePodSpecWithProxy(podSpec *corev1.PodSpec, containerNames []string, httpProxy, httpsProxy, noProxy string) error {
+	hasProxy := len(httpsProxy) > 0 || len(httpProxy) > 0 || len(noProxy) > 0
```

Comment:
If only NoProxy was set, I'm not sure that's going to work correctly...

Reply:
> If only NoProxy was set, I'm not sure that's going to work correctly...

Empty is the same as unset.  See https://github.com/golang/net/blob/master/http/httpproxy/proxy.go#L91-L107

Reply:
+1